scipy/stats/_axis_nan_policy.py

```diff
@@ -226,6 +226,25 @@ def _check_empty_inputs(samples, axis):
     return output
 
 
+def _add_reduced_axes(res, reduced_axes, n_outputs,
+                      result_object):
+    res_expanded = []
+    if n_outputs == 1:
+        # Since NumPy 1.18, `expand_dims` supports tuple axis. As SciPy
+        # supports any NumPy >= 1.17, we can't guarantee support for tuple
+        # axis by NumPy. This for loop can be removed when support for
+        # NumPy 1.17 is dropped.
+        for a in reduced_axes:
```

Comment:
I'd expect this could run into trouble if `reduced_axes` is not sorted. I'm not certain there is a problem, but please consider it. If there is a problem, the tests wouldn't catch it because they are all sorted in an order this works for. The `(1, -1)` test comes closest to catching it, but doesn't because `-1` is equivalent to `3`.

Reply:
> I'd expect this could run into trouble if `reduced_axes` is not sorted.

You are right. Thanks for catching this. `_broadcast_shapes` function already sorted and validated axis. I have separated that part into a `_process_axis` function and used it sort `reduced_axis`.

One catch is that it is possible that the user didn't pass any samples (which leads to an error down the line), in which case, `samples` array is empty and processing axis doesn't make any sense in that case. To handle this, I have taken as input the shapes of the samples and, if it is empty, we just return the axis as is. I thought of passing the `n_dims` parameter instead, but if there are no samples, there is no way to calculate it (alternative is to pass a invalid value like `-1` to indicate failure).